### PR TITLE
Fix condition not set occasionally

### PR DESCRIPTION
**Pull request by zhujian7**
The SecretCreated type condition is only added when the secret is created the first time, if the condition update operation fails, the condition will not be added in the next reconciliation.
This PR makes the operation of updating conditions retriable

### CODE REVIEW

**Comment by zhujian7** [comment]
cc @elgnay

File: pkg/addon/agent/controller/token.go [hunk 1]
```diff
@@ -82,86 +83,119 @@ func (r *TokenReconciler) Reconcile(ctx context.Context, request reconcile.Reque
...
+		conditions = append(conditions, metav1.Condition{
+			Type:               authv1beta1.ConditionTypeTokenReported,
+			Status:             metav1.ConditionTrue,
+			Reason:             "TokenReported",
+			LastTransitionTime: now,
+		})
+	}
+
+	// after sync func succeeds, the secret must exist, add the conditions if not exist
+	if meta.FindStatusCondition(managed.Status.Conditions, authv1beta1.ConditionTypeSecretCreated) == nil {
```
Comment (qiujian16) [inline pkg/addon/agent/controller/token.go:113]: I think you can use the helper in meta to directly set the condition?

Comment (zhujian7) [reply pkg/addon/agent/controller/token.go]: I think these 2 conditions are similar, can we only keep one and delete another one?

Comment (zhujian7) [reply pkg/addon/agent/controller/token.go]: @qiujian16 ptal

**Comment by zhujian7** [comment]
@elgnay please help take a look

**Pull request merged**